go/interpreter/lfvm/instructions_test.go

```diff
@@ -1192,6 +1192,104 @@ func TestGenericCreate_MaxInitCodeSizeIsNotCheckedBeforeShanghai(t *testing.T) {
 	}
 }
 
+func TestGenericCreate_ReportsErrors(t *testing.T) {
+
+	u64Overflow := new(uint256.Int).Add(uint256.NewInt(math.MaxUint64), uint256.NewInt(math.MaxUint64))
+	one := uint256.NewInt(1)
+
+	tests := map[string]struct {
+		offset, size  uint256.Int
+		kind          tosca.CallKind
+		revision      tosca.Revision
+		expectedError error
+	}{
+		"offset overflow": {
+			offset:        *u64Overflow,
+			size:          *one,
+			expectedError: errOverflow,
+		},
+		"size overflow": {
+			offset:        *uint256.NewInt(0),
+			size:          *u64Overflow,
+			expectedError: errOverflow,
+		},
+		"size+offset overflow": {
+			offset:        *uint256.NewInt(math.MaxUint64 - 4),
+			size:          *uint256.NewInt(5),
+			expectedError: errOverflow,
+		},
```

Comment:
Some, if not all of these overflow checks may get obsolete once #838 is merged. This PR should be sequenced after #838.